src/native-state-drm.cpp

```diff
@@ -441,10 +450,13 @@ NativeStateDRM::fb_get_from_bo(gbm_bo* bo)
 
     unsigned int width = gbm_bo_get_width(bo);
     unsigned int height = gbm_bo_get_height(bo);
-    unsigned int stride = gbm_bo_get_stride(bo);
-    unsigned int handle = gbm_bo_get_handle(bo).u32;
+    unsigned int handles[4] = { gbm_bo_get_handle(bo).u32, 0, };
+    unsigned int pitches[4] = { gbm_bo_get_stride(bo), 0, };
+    unsigned int offsets[4] = { 0, };
```

Comment:
These look good for the vast majority of DRM formats that we are expected to get. However, there may be some cases/platforms where we end up with some kind of DRM format that requires multiplane support, and thus properly populated handles/pitches/offsets. Since we are fixing format support and switching to AddFB2, let's also populate these arrays fully to get the full benefit of this change (e.g., with `gbm_bo_get_handle_for_plane`, `gbm_bo_get_stride_for_plane`, `gbm_bo_get_offset`).

Reply:
those APIs are new, not all gbm versions support them. maybe we need to add something like the weston's gbm version check and provide both ways(single-plane and multi-plane):
        if dep_gbm.version().version_compare('>= 17.1')
                config_h.set('HAVE_GBM_MODIFIERS', '1')
        endif